src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertorTest.java

```diff
@@ -162,6 +167,25 @@ public void canConvertAValidSecretWithNoIamMfa() throws Exception {
             assertThat("credential accessKey is mapped correctly", credential.getAccessKey(), is(accessKey));
             assertThat("credential secretKey is mapped correctly", credential.getSecretKey().getPlainText(), is(secretKey));
             assertThat("credential iamRoleArn is mapped correctly", credential.getIamRoleArn(), is(iamRoleArn));
+            assertThat("credential iamExternalId is mapped correctly", credential.getIamExternalId(), is(iamExternalId));
+        }
+    }
+    
+    @Test
+    public void canConvertAValidSecretWithNoIamExternalId() throws Exception {
+        AWSCredentialsConvertor convertor = new AWSCredentialsConvertor();
+
+        try (InputStream is = get("validMissingIamExternalId.yaml")) {
+            Secret secret = Serialization.unmarshal(is, Secret.class);
+            assertThat("The Secret was loaded correctly from disk", notNullValue());
+            AWSCredentialsImpl credential = convertor.convert(secret);
+            assertThat(credential, notNullValue());
+            assertThat("credential id is mapped correctly", credential.getId(), is("a-test-aws"));
+            assertThat("credential description is mapped correctly", credential.getDescription(), is(emptyString()));
+            assertThat("credential scope is mapped correctly", credential.getScope(), is(CredentialsScope.GLOBAL));
+            assertThat("credential accessKey is mapped correctly", credential.getAccessKey(), is(accessKey));
+            assertThat("credential secretKey is mapped correctly", credential.getSecretKey().getPlainText(), is(secretKey));
+            assertThat("credential iamRoleArn is mapped correctly", credential.getIamRoleArn(), is(iamRoleArn));
```

Comment:
Should you add a check that the external id check 
 is `null` here?

Reply:
@jtnord can you elaborate or add an example? None of the other missing item validations are explicitely checking if that value is null

Reply:
https://github.com/jenkinsci/kubernetes-credentials-provider-plugin/blob/master/src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/OpenstackCredentialv3ConvertorTest.java#L136